openfe/protocols/openmm_rfe/equil_rfe_methods.py

```diff
@@ -527,21 +527,16 @@ def run(self, *, dry=False, verbose=True,
         )
 
         #  b. Write out a PDB containing the subsampled hybrid state
-        A_atoms = {hybrid_factory.old_to_hybrid_atom_map[i]
-                   for i in ligand_mappings['old_mol_indices']}
-        B_atoms = {hybrid_factory.new_to_hybrid_atom_map[i]
-                   for i in ligand_mappings['new_mol_indices']}
-        A_unique = list(A_atoms - B_atoms)
-        AB_core = list(A_atoms & B_atoms)
-        B_unique = list(B_atoms - A_atoms)
+        A_unique = list(hybrid_factory._atom_classes['unique_old_atoms'])
+        AB_core = list(hybrid_factory._atom_classes['core_atoms'])
+        B_unique = list(hybrid_factory._atom_classes['unique_new_atoms'])
+        protein = []  # TODO
```

Comment:
This isn't where you should be tracking your protein atoms imho. It's a PDB output so you have residue names, those should be used.

As previously mentioned - tracking crystallographic waters is also not an amazing idea since water exchange is a thing.